internal/services/checker/deepscan/ast_utils.go

```diff
@@ -97,8 +97,8 @@ func parseRecursive(
 			return nil, fmt.Errorf("failed parse '%s': %w", analysePackage, err)
 		}
 
-		for packageID, astPackage := range found {
-			pkgs[packageID] = astPackage
+		for _, astPackage := range found {
+			pkgs[analysePackage] = astPackage
```

Comment:
analysePackage is not unique either.
I think something like `fmt.Sprintf("%s_%s", analysePackage, packageID)` should be here.

because many packages can be in one directory (common case is tests):
```
analysePackage = /some/project/internal/code/repository

found = [2]Package
- repository
- repository_test
```

Reply:
Missed that case, thank you. Replaced analysePackage with your variant